File: 프로그래머스/lv1/42840. 모의고사/모의고사.swift
import Foundation

func solution(_ answers:[Int]) -> [Int] {
    var answer:[Int] = []
    var maxValue = 0
    
    // 노트1. repeating을 배열 기준으로 할 수 있다.
    var firstArr = Array(repeating: [1, 2, 3, 4, 5], count:10000/5).flatMap { $0 }
    var secondArr = Array(repeating: [2, 1, 2, 3, 2, 4, 2, 5], count:10000/8).flatMap { $0 }
    var thirdArr = Array(repeating: [3, 3, 1, 1, 2, 2, 4, 4, 5, 5], count:10000/10).flatMap { $0 }
    
    var first = 0, second = 0, third = 0
    
    for i in 0..<answers.count {
        if firstArr[i] == answers[i] {
            first += 1
        }
        
        if secondArr[i] == answers[i] {
            second += 1
        }
        
        if thirdArr[i] == answers[i] {
            third += 1
        }
    }
    
    maxValue = [first, second, third].max()!
    
    return [(1, first), (2, second), (3, third)].filter { $0.1 == maxValue }.map { $0.0 }
}

File: 프로그래머스/lv1/42840. 모의고사/모의고사.py
def solution(answers):
    one = [1, 2, 3, 4, 5] * 2000
    two = [2, 1, 2, 3, 2, 4, 2, 5] * 1250
    three = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5] * 1000
    
    count = [0] * 3
    
    for i in range(len(answers)):
        if answers[i] == one[i]:
            count[0] += 1
        if answers[i] == two[i]:
            count[1] += 1
        if answers[i] == three[i]:
            count[2] += 1
    _max = max(count)
    answer = []
    
    for i in range(3):
        if count[i] == _max:
            answer.append(i+1)
    
    return answer